lib/std/os/linux/loongarch64.zig

```diff
@@ -0,0 +1,218 @@
+        : [number] "{$r11}" (@intFromEnum(number)),
+          [arg1] "{$r4}" (arg1),
+          [arg2] "{$r5}" (arg2),
+          [arg3] "{$r6}" (arg3),
+          [arg4] "{$r7}" (arg4),
+          [arg5] "{$r8}" (arg5),
+          [arg6] "{$r9}" (arg6),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+const CloneFn = *const fn (arg: usize) callconv(.C) u8;
+
+pub extern fn clone(func: CloneFn, stack: usize, flags: u32, arg: usize, ptid: *i32, tls: usize, ctid: *i32) usize;
+
+pub const restore = restore_rt;
+
+pub fn restore_rt() callconv(.Naked) noreturn {
+    asm volatile (
+        \\ or $a7, $zero, %[number]
+        \\ syscall 0
+        :
+        : [number] "r" (@intFromEnum(SYS.rt_sigreturn)),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+pub const LOCK = struct {
+    pub const SH = 1;
+    pub const EX = 2;
+    pub const UN = 8;
+    pub const NB = 4;
+};
+
+pub const blksize_t = i32;
+pub const nlink_t = u32;
+pub const time_t = isize;
+pub const mode_t = u32;
+pub const off_t = isize;
+pub const ino_t = usize;
+pub const dev_t = usize;
+pub const blkcnt_t = isize;
+
+pub const timeval = extern struct {
+    tv_sec: time_t,
+    tv_usec: i64,
+};
+
+pub const O = struct {
+    pub const CREAT = 0o100;
+    pub const EXCL = 0o200;
+    pub const NOCTTY = 0o400;
+    pub const TRUNC = 0o1000;
+    pub const APPEND = 0o2000;
+    pub const NONBLOCK = 0o4000;
+    pub const DSYNC = 0o10000;
+    pub const SYNC = 0o4010000;
+    pub const RSYNC = 0o4010000;
+    pub const DIRECTORY = 0o200000;
+    pub const NOFOLLOW = 0o400000;
+    pub const CLOEXEC = 0o2000000;
+
+    pub const ASYNC = 0o20000;
+    pub const DIRECT = 0o40000;
+    pub const LARGEFILE = 0o100000;
+    pub const NOATIME = 0o1000000;
+    pub const PATH = 0o10000000;
+    pub const TMPFILE = 0o20200000;
+    pub const NDELAY = NONBLOCK;
+};
+
+pub const F = struct {
+    pub const DUPFD = 0;
+    pub const GETFD = 1;
+    pub const SETFD = 2;
+    pub const GETFL = 3;
+    pub const SETFL = 4;
+    pub const GETLK = 5;
+    pub const SETLK = 6;
+    pub const SETLKW = 7;
+    pub const SETOWN = 8;
+    pub const GETOWN = 9;
+    pub const SETSIG = 10;
+    pub const GETSIG = 11;
+
+    pub const RDLCK = 0;
+    pub const WRLCK = 1;
+    pub const UNLCK = 2;
+
+    pub const SETOWN_EX = 15;
+    pub const GETOWN_EX = 16;
+
+    pub const GETOWNER_UIDS = 17;
+};
+
+pub const MAP = struct {
+    /// stack-like segment
+    pub const GROWSDOWN = 0x0100;
+    /// ETXTBSY
+    pub const DENYWRITE = 0x0800;
+    /// mark it as an executable
+    pub const EXECUTABLE = 0x1000;
+    /// pages are locked
+    pub const LOCKED = 0x2000;
+    /// don't check for reservations
+    pub const NORESERVE = 0x4000;
+};
+
+pub const VDSO = struct {
+    pub const CGT_SYM = "__kernel_clock_gettime";
+    pub const CGT_VER = "LINUX_5.10";
+};
+
+pub const mcontext_t = extern struct {
```

Comment:
What's the source for this and `ucontext_t` below?

Reply:
They are from:

https://elixir.bootlin.com/linux/v5.19/source/arch/loongarch/include/uapi/asm/sigcontext.h
https://elixir.bootlin.com/linux/v5.19/source/arch/loongarch/include/uapi/asm/ucontext.h

I'm not sure whether I added these fields correctly.

Reply:
The `mcontext_t` name is odd, but I see that's just following established convention in the arch bits files, so let's ignore that.

The fields aren't correct, though. It should probably look something like this:

```zig
pub const mcontext_t = extern struct {
    pc: u64,
    regs: [32]u64,
    flags: u32,
    extcontext: [0]u64 align(16),
};
```

The last field actually matters because its alignment causes tail padding in the structure.